src/main/java/com/google/sps/placeGuide/repository/impl/DatastorePlaceGuideRepository.java

```diff
@@ -117,53 +115,99 @@ public PlaceGuide getPlaceGuide(long placeGuideId) {
+    Filter northBoundFilter =
+        new FilterPredicate(
+            COORDINATE_PROPERTY, FilterOperator.LESS_THAN_OR_EQUAL, northEastCorner);
     Filter creatorFilter =
         new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId);
-    Filter queryFilter = CompositeFilterOperator.and(publicityFilter, mapAreaFilter, creatorFilter);
+    Filter queryFilter =
+        CompositeFilterOperator.and(
+            publicityFilter, southBoundFilter, northBoundFilter, creatorFilter);
     Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
-    return getPlaceGuidesList(query);
+    return removePlaceGuidesOutsideLongitudeBounds(
+        getPlaceGuidesList(query), southWestCorner.getLongitude(), northEastCorner.getLongitude());
   }
 
   @Override
   public List<PlaceGuide> getCreatedPrivatePlaceGuidesInMapArea(
       String creatorId, GeoPt northEastCorner, GeoPt southWestCorner) {
     Filter publicityFilter = new FilterPredicate(IS_PUBLIC_PROPERTY, FilterOperator.EQUAL, false);
-    Filter mapAreaFilter =
-        new StContainsFilter(
-            COORDINATE_PROPERTY, new GeoRegion.Rectangle(southWestCorner, northEastCorner));
+    Filter southBoundFilter =
+        new FilterPredicate(
+            COORDINATE_PROPERTY, FilterOperator.GREATER_THAN_OR_EQUAL, southWestCorner);
+    Filter northBoundFilter =
+        new FilterPredicate(
+            COORDINATE_PROPERTY, FilterOperator.LESS_THAN_OR_EQUAL, northEastCorner);
     Filter creatorFilter =
         new FilterPredicate(CREATOR_ID_PROPERTY, FilterOperator.EQUAL, creatorId);
-    Filter queryFilter = CompositeFilterOperator.and(publicityFilter, mapAreaFilter, creatorFilter);
+    Filter queryFilter =
+        CompositeFilterOperator.and(
+            publicityFilter, southBoundFilter, northBoundFilter, creatorFilter);
     Query query = new Query(ENTITY_KIND).setFilter(queryFilter);
-    return getPlaceGuidesList(query);
+    return removePlaceGuidesOutsideLongitudeBounds(
+        getPlaceGuidesList(query), southWestCorner.getLongitude(), northEastCorner.getLongitude());
+  }
+
+  private List<PlaceGuide> removePlaceGuidesOutsideLongitudeBounds(
+      List<PlaceGuide> placeGuides, float westBound, float eastBound) {
+    List<PlaceGuide> placeGuidesCopy = new ArrayList<>(placeGuides);
+    placeGuidesCopy.removeIf(
+        placeGuide ->
+            !isBetweenLongitudeBounds(
+                placeGuide.getCoordinate().getLongitude(), westBound, eastBound));
+    return placeGuidesCopy;
+  }
+
+  private boolean isBetweenLongitudeBounds(float longitude, float westBound, float eastBound) {
+    if (westBound < eastBound) {
+      return westBound <= longitude && longitude <= eastBound;
+    } else {
+      // The bounded area crosses the International Date Line(located at longitude 180/-180).
+      // In this case, the westBound is positive and the eastBound is negative.
+      if (longitude > 0) {
```

Comment:
maybe change it to else if statement?

Reply:
Wouldn't do that. You are right that it would make the code shorter, but I think this structure offers a better readability. This way, it is clear that I want to handle separately the cases when the rectangle crosses/it doesn't cross the IDL.